### PR TITLE
created a modal window

### LINKED ISSUE #48: ui/modal
Create a modal window

### PR DESCRIPTION
**Pull request by aavdienko**
@Stanislavussov , hi! Please checkout my modal window component. Looking forward to get your feedback and have this request merged.

### CODE REVIEW

File: src/components/ui/modal/modal.module.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
+.DialogContent {
```
Comment (Stanislavussov) [inline src/components/ui/modal/modal.module.scss:1]: what approach of styling do u use? Didnt even meet class starting from capital letter

File: src/components/ui/modal/modal.module.scss [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  align-items: center;
+  justify-content: space-between;
+  padding: 0 24px;
+}
+
+.headerWithoutCLose {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: 18px 24px;
```
Comment (Stanislavussov) [inline src/components/ui/modal/modal.module.scss:34]: try to avoid `px` use `rem`

File: src/components/ui/modal/modal.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { FC, ReactNode, useState } from 'react'
+
```
Comment (Stanislavussov) [inline src/components/ui/modal/modal.tsx:2]: i think this spaces i redundant and make file longer

File: src/components/ui/modal/modal.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            <main className={s.main}>{children}</main>
+            <footer className={showCloseBtn ? s.footer : s.footerWithoutClose}>
+              {showCloseBtn && (
+                <Dialog.Close asChild>
+                  <Button variant="secondary">Cancel</Button>
+                </Dialog.Close>
+              )}
+              <Dialog.Close
+                asChild
+                onClick={() => {
```
Comment (Stanislavussov) [inline src/components/ui/modal/modal.tsx:58]: use separate callback as it go

File: src/components/ui/modal/modal.module.scss [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  display: flex;
+  flex-direction: column;
+  gap: 24px;
+  align-items: start;
+
+  padding: 24px;
+}
+
+.footer {
+  display: flex;
```
Comment (Stanislavussov) [inline src/components/ui/modal/modal.module.scss:47]: for flex i would create in some future separate component like `FlexBox` to manage easier this cases

File: src/components/ui/modal/modal.tsx (hunk 3 above)
Comment (aavdienko) [reply src/components/ui/modal/modal.tsx:2]: 100% agree, but it's prettier and eslint preset, so we have to use this...

File: src/components/ui/modal/modal.module.scss (hunk 2 above)
Comment (aavdienko) [reply src/components/ui/modal/modal.module.scss:34]: Will try to use remS, I did it only once as of now

**Pull request merged**